Terminal.Gui/Drawing/Color.cs

```diff
@@ -110,51 +132,140 @@ public TrueColor (int red, int green, int blue)
+		/// Converts a <see cref="Color"/> to a <see cref="TrueColor"/> using a default mapping.
+		/// </summary>
+		/// <param name="consoleColor">The <see cref="Color"/> to convert.</param>
+		/// <returns></returns>
+		public static TrueColor FromConsoleColor (Color consoleColor)
+		{
+			switch (consoleColor) {
+			case Color.Black: return new TrueColor (0, 0, 0);
+			case Color.Blue: return new TrueColor (0, 0, 0x80);
+			case Color.Green: return new TrueColor (0, 0x80, 0);
+			case Color.Cyan: return new TrueColor (0, 0x80, 0x80);
+			case Color.Red: return new TrueColor (0x80, 0, 0);
+			case Color.Magenta: return new TrueColor (0x80, 0, 0x80);
+			case Color.Brown: return new TrueColor (0xC1, 0x9C, 0x00); // TODO confirm this
+			case Color.Gray: return new TrueColor (0xC0, 0xC0, 0xC0);
+			case Color.DarkGray: return new TrueColor (0x80, 0x80, 0x80);
+			case Color.BrightBlue: return new TrueColor (0, 0, 0xFF);
+			case Color.BrightGreen: return new TrueColor (0, 0xFF, 0);
+			case Color.BrightCyan: return new TrueColor (0, 0xFF, 0xFF);
+			case Color.BrightRed: return new TrueColor (0xFF, 0, 0);
+			case Color.BrightMagenta: return new TrueColor (0xFF, 0, 0xFF);
+			case Color.BrightYellow: return new TrueColor (0xFF, 0xFF, 0);
+			case Color.White: return new TrueColor (0xFF, 0xFF, 0xFF);
+			default: return new TrueColor (0x80, 0, 0); // TODO getting unknown color value (-1) during JSON serialization
+								    // Don't know how to map it
+			};
+		}
+
+		/// <summary>
+		/// Converts the provided <see cref="TrueColor"/> to <see cref="Color"/> using a default mapping.
+		/// </summary>
+		/// <param name="trueColor"></param>
+		/// <returns></returns>
+		public static Color ToConsoleColor (TrueColor trueColor)
+		{
+			return TrueColorToConsoleColorMap.OrderBy (kv => CalculateDistance (kv.Key, trueColor)).First ().Value;
 		}
 
-		private float CalculateDistance (TrueColor color1, TrueColor color2)
+		private static float CalculateDistance (TrueColor color1, TrueColor color2)
 		{
 			// use RGB distance
 			return
 				Math.Abs (color1.Red - color2.Red) +
 				Math.Abs (color1.Green - color2.Green) +
 				Math.Abs (color1.Blue - color2.Blue);
 		}
+
+		/// <inheritdoc/>
+		public static bool operator == (TrueColor left, TrueColor right)
+		{
+			return left.Equals (right);
+		}
+
+		/// <inheritdoc/>
+		public static bool operator != (TrueColor left, TrueColor right)
+		{
+			return !left.Equals (right);
+		}
+
+		/// <inheritdoc/>
+		public override bool Equals (object obj)
+		{
+			return obj is TrueColor other && Equals (other);
+		}
+
+		/// <inheritdoc/>
+		public bool Equals (TrueColor other)
+		{
+			return
+				Red == other.Red &&
+				Green == other.Green &&
+				Blue == other.Blue;
+		}
+
+		/// <inheritdoc/>
+		public override int GetHashCode ()
+		{
+			return HashCode.Combine (Red, Green, Blue);
+		}
+
+		/// <inheritdoc/>
+		public override string ToString ()
+		{
+			return $"#{Red:X2}{Green:X2}{Blue:X2}";
```

Comment:
Defaults the serialization to use RGB HexCode. Is this what you want? Color reads from rgb(...). I added support to parse TrueColor from rgb(...) or #RRGGBB or #RGB.